cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/ResultsBinding.java

```diff
@@ -102,12 +103,13 @@ public String resolveResult(Map<String, Value> inputs,
                 try {
                     Value expressionResult = scriptEvaluator.evalExpr(expression, scriptContext, systemProperties, result.getFunctionDependencies());
                     Boolean evaluatedResult;
-                    if (expressionResult.get() instanceof Integer) {
-                        evaluatedResult = (Integer) expressionResult.get() != 0;
+                    String expressionResultValue = (String) expressionResult.get();
+                    if (NumberUtils.isNumber(expressionResultValue)) {
```

Comment:
I think in results case, the script eval should return a boolean as before (no change), typing is only for variables